### DISCUSSION TITLE
Automatically detect Markdown mail indentation

### DISCUSSION THREAD

**Discussion by bram-pkg** [laravel/framework#40359] [Ideas]
### Description:

When formatting a file according to the same logic other views (i.e. indenting after a `@component` tag), the Markdown is rendered as a code block. I think it would be more user friendly if the Markdown mailer automatically detects the minimum indent level, and strips those whitespaces. 

```md
@component('mail::message')
    # Welcome
    
    Body of email
    
    @component('mail::button', ['url' => $url])
        Open PDF
    @endcomponent
    
    With kind regards,
    
    {{ config('app.name') }}
@endcomponent
```

Would detect the 4-space indent and trim them.

**Comment by clicktodev** [comment]
Definitely agree with this.
Related to #40356 and #43395
